### PR TITLE
testing: Add simple debug test program

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
It is a simple test case of the NuttX debugging facilities (breakpoint and watchpoint).

See also: https://github.com/apache/nuttx/pull/12549
## Impact
New test case
## Testing
STM32F411

### CODE REVIEW

File: testing/debug/Kconfig [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#
+# For a description of the syntax of this configuration file,
+# see the file kconfig-language.txt in the NuttX tools repository.
+#
+
+config TESTING_DEBUG
+	tristate "Debug test"
+	default n
+	depends on ARCH_HAVE_DEBUG
```
Comment (xiaoxiang781216) [inline testing/debug/Kconfig:9]: remove, it's enough to depends on LIB_GDBSTUB

File: testing/debug/debug.c [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+      if (triggered)
+        {
+          printf("Pass: Read watchpoint at %p triggered\n", addr);
+        }
+      else
+        {
+          gdb_debugpoint_remove(test_type, addr, size);
+          printf("ERROR: Read watchpoint at %p not triggered\n", addr);
+          ret = -1;
```
Comment (xiaoxiang781216) [inline testing/debug/debug.c:173]: should we remove ALL `ret = -1` and `ret = 0`

File: testing/debug/Kconfig (hunk 1 above)
Comment (no1wudi) [reply testing/debug/Kconfig:9]: No, it's possible that LIB_GDBSTUB enabled but ARCH_HAVE_DEBUG disabled.

File: testing/debug/debug.c (hunk 2 above)
Comment (no1wudi) [reply testing/debug/debug.c]: OK, removed

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**